pubsub/api/PubsubTest/PubsubTest.cs

```diff
@@ -42,15 +42,60 @@ public class PubsubTest : IDisposable
             Command = "Pubsub"
         };
 
+        /// <summary>
+        /// Handle a special race condition that can occur when deleting
+        /// something:
+        /// 1. Delete request times out.
+        /// 2. Delete operation continues on server and succeeds.
+        /// 3. Later requests to delete the same entity see NotFound error.
+        /// </summary>
+        /// <param name="delete">The delete operation to run.</param>
+        /// <returns>An action to run inside Eventually().</returns>
+        Action HandleDeleteRace(Action delete)
+        {
+            bool sawTimeout = false;
+            return () =>
+            {
+                if (!sawTimeout)
+                {
+                    try
+                    {
+                        delete();
+                    }
+                    catch (Grpc.Core.RpcException e)
+                    when (e.Status.StatusCode == StatusCode.DeadlineExceeded)
+                    {
+                        sawTimeout = true;
+                        throw;
+                    }
+                }
+                else
+                {
+                    try
+                    {
+                        delete();
+                    }
+                    catch (Grpc.Core.RpcException e)
+                    when (e.Status.StatusCode == StatusCode.NotFound)
+                    {
+                        // Earlier timeout request that deleted the thing
+                        // actually succeeded on the server.
+                    }
+                }
+            };
+        }
+
         void IDisposable.Dispose()
         {
             foreach (string topicId in _tempTopicIds)
             {
-                Eventually(() => Run("deleteTopic", _projectId, topicId));
+                Eventually(HandleDeleteRace(() =>
+                    Run("deleteTopic", _projectId, topicId)));

```

Comment:
What you can do is catch the exception you throw in "HandleDeleteRace" here and if exception is thrown, you can do:
Assert.True(false, "Cleanup of the topic named xxx has failed. More details: xxx")
(Apparently, xunit doesn't support Assert.Fail...)

The reason I think this is better than just let the exception propagate:

1) When looking at the failing test (even though it's test cleanup that failed, corresponding test is marked as failed...) the message will say the real reason of what happened as opposed to "Exception of type xxx was thrown". Will make it easier to troubleshoot even without delving into the code.
2) It will look as best effort to follow best practices within xunit implementation constraints.